### DISCUSSION TITLE
How to generate documentation for Magefiles?

### DISCUSSION THREAD

**Discussion by mloskot** [magefile/mage#493] [Q&A]
Is there any way to generate Markdown or HTML with documentation for Mage exports in `magefiles/*.go`?

I have tried godoc, but it does not seem to generate anything for definitions in Magefiles.

**Comment by sheldonhull** [comment]
I wrote this and it seemed to document the functions correctly. 

https://github.com/sheldonhull/magetools/tree/main/docgen

**Reply by mloskot** [reply]
Nice, I will try it out, thanks!
